spyder/widgets/comboboxes.py

```diff
@@ -135,6 +135,8 @@ class PatternComboBox(BaseComboBox):
     def __init__(self, parent, items=None, tip=None,
                  adjust_to_minimum=True):
         BaseComboBox.__init__(self, parent)
+        if hasattr(self.lineEdit(), 'setClearButtonEnabled'):  # only Qt >= 5.2

```

Comment:
Given that this only works for `Qt > = 5.2`, you need to enclose the lines you added in

```python
if PYQT5:
    ...
```

where `PYQT5` can be imported like this


```python
from qtpy import PYQT5
```

Reply:
The `if hasattr()` serves exactly that purpose: Only use this feature if available. This should work without explicity checking a Qt version.

Also, I would assume that a simple `if PYQT5` would be `True` for  5.0 <= Qt < 5.2, but they still don't have the clear button.

Reply:
Sorry, you're right, `hasattr` should be enough.

Besides, we only support `Qt 5.2+`

Reply:
> Besides, we only support Qt 5.2+

In that case, just `self.lineEdit().setClearButtonEnabled(True)` would be enough.

The `if` is unneccesary clutter then. Should I remove it for brevity and clarity of the code?

Reply:
Nop, because we also support PyQt4.